src/clp_ffi_js/ir/LogViewerEvent.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef CLP_FFI_JS_IR_LOG_VIEWER_EVENT_HPP
+#define CLP_FFI_JS_IR_LOG_VIEWER_EVENT_HPP
+
+#include <clp/ir/LogEvent.hpp>
+#include <utility>
+
+namespace clp_ffi_js::ir {
+
+/**
+ * A class derived from LogEvent with an additional member for log level.
+ * @tparam encoded_variable_t The type of encoded variables in the event
+ */
+template <typename encoded_variable_t>
+class LogViewerEvent : public clp::ir::LogEvent<encoded_variable_t> {
```

Comment:
(As in we might want to be less specific about the Log Viewer which is a final product. ) If done correctly, in the future we can release clp-ffi-js as a Node.js library and the extended attributes like the log level are still useful, right?

Reply:
Can we call it `LogEventWithLevel`? I prefer to pick a strangely specific name because eventually, once we move to the next IR format, LogEvent should have a field which is its level. So by picking and odd name now, I think it will hint to the reader that this code has something odd about it and they will look more into the docstring (where we can explain the situation).